### PR TITLE
LT-1114 add basic info for types of goods

### PR DESCRIPTION
**Pull request by 5pence**
review

### CODE REVIEW

File: apply_for_a_licence/forms/initial.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 from libraries.forms.components import Form, ArrayQuestion, InputType, DetailComponent, Question, Option, Section
+from conf.settings import env
```
Comment (cstobart) [inline apply_for_a_licence/forms/initial.py]: is this import needed

File: apply_for_a_licence/urls.py [hunk 2]
```diff
@@ -12,7 +14,10 @@
     # ex: /<uuid:pk>/overview/
     path('<uuid:pk>/overview/', common.Overview.as_view(), name='overview'),
```
Comment (cstobart) [inline apply_for_a_licence/urls.py]: blank line

File: apply_for_a_licence/views/common.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 from core.builtins.custom_tags import get_string
 from core.services import get_units, get_sites_on_draft, get_external_locations_on_draft
 from drafts.services import post_drafts, get_draft, get_draft_goods, post_draft_preexisting_goods, submit_draft, \
-    delete_draft, post_end_user
+    delete_draft, post_end_user, get_draft_goodstype
```
Comment (cstobart) [inline apply_for_a_licence/views/common.py]: rename app/views/services to <something>_goods_type from <something>_goodstype

File: drafts/services.py [hunk 4]
```diff
@@ -38,6 +38,11 @@ def get_draft_goods(request, pk):
     return data.json(), data.status_code
 
 
+def get_draft_goodstype(request, pk):
```
Comment (cstobart) [inline drafts/services.py]: rename to get_draft_goods_type

File: goodstype/apps.py [hunk 5]
```diff
@@ -0,0 +1,5 @@
+from django.apps import AppConfig
+
+
+class GoodstypeConfig(AppConfig):
```
Comment (cstobart) [inline goodstype/apps.py]: rename GoodsTypeConfig

File: goodstype/services.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
+from urllib.parse import urlencode
+
+from conf.client import get, post, put, delete
+from conf.constants import GOODSTYPE_URL
+
+
+def get_goodstypes(request, params=None):
```
Comment (cstobart) [inline goodstype/services.py]: rename all of the services

File: goodstype/urls.py [hunk 7]
```diff
@@ -0,0 +1,13 @@
+from django.urls import path
+
+from . import views
+
+app_name = 'goodstype'
```
Comment (cstobart) [inline goodstype/urls.py]: app_name = 'goods_type'

File: goodstype/urls.py [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+urlpatterns = [
+    # ex: /goodstype/
+    path('<uuid:pk>', views.GoodsType.as_view(), name='goodstype'),
```
Comment (cstobart) [inline goodstype/urls.py]: name='goods_type'

File: goodstype/views.py [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    def get(self, request, **kwargs):
+        return form_page(request, forms.form)
+
+    def post(self, request, **kwargs):
+        copied_post = request.POST.copy()
+        copied_post['content_type'] = DRAFT_CONTENT_TYPE_ID
+        copied_post['object_id'] = kwargs.get('pk')
+        data, status_code = post_goodstype(request, copied_post)
+
+        if status_code == 400:
```
Comment (cstobart) [inline goodstype/views.py]: if status_code != 200:

is better

File: goodstype/views.py [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    def get(self, request, **kwargs):
+        return form_page(request, forms.form)
+
+    def post(self, request, **kwargs):
+        copied_post = request.POST.copy()
+        copied_post['content_type'] = DRAFT_CONTENT_TYPE_ID
+        copied_post['object_id'] = str(kwargs.get('pk'))
+        data, status_code = post_goodstype(request, copied_post)
+
+        if status_code == 400:
```
Comment (cstobart) [inline goodstype/views.py]: if status_code != 200:

File: templates/apply_for_a_licence/goods/index.html [hunk 11]
```diff
@@ -11,12 +11,13 @@
...
 		<h2 class="govuk-heading-l">{{ title }}</h2>
 		<div class="lite-app-bar--right">
-			<a href="{% url 'apply_for_a_licence:preexisting_good' draft_id %}" role='button' draggable='false' class='govuk-button'>Add from organisations goods</a>
+			{% block right_button %}<a href="{% url 'apply_for_a_licence:preexisting_good' draft_id %}" role='button' draggable='false' class='govuk-button'>Add from organisations goods</a>{% endblock %}
 		</div>
 	</div>
 
 	{% if data.goods|length %}
 		{% for good in data.goods %}
+            {% block good %}
```
Comment (cstobart) [inline templates/apply_for_a_licence/goods/index.html]: indentation?

File: templates/apply_for_a_licence/overview.html [hunk 12]
```diff
@@ -93,9 +93,15 @@ <h2 class="govuk-heading-l">
 <div class="lite-section {% if errors|key_value:'goods' %}govuk-form-group--error{% endif %}">
 	<div class="lite-section-header">
 		<div>
-			<a id="goods" href="{% url 'apply_for_a_licence:goods' draft.id %}" class="lite-overview-section-header govuk-link govuk-link--no-visited-state">
-				Goods
-			</a>
+            {% if draft.licence_type == 'open_licence' %}
```
Comment (cstobart) [inline templates/apply_for_a_licence/overview.html]: indentations?

File: apply_for_a_licence/forms/initial.py (hunk 1 above)
Comment (antonhughes) [reply apply_for_a_licence/forms/initial.py]: removed

File: apply_for_a_licence/urls.py (hunk 2 above)
Comment (antonhughes) [reply apply_for_a_licence/urls.py]: removed

File: apply_for_a_licence/views/common.py (hunk 3 above)
Comment (antonhughes) [reply apply_for_a_licence/views/common.py]: renamed

File: drafts/services.py (hunk 4 above)
Comment (antonhughes) [reply drafts/services.py]: done

File: goodstype/apps.py (hunk 5 above)
Comment (antonhughes) [reply goodstype/apps.py]: done

File: goodstype/services.py (hunk 6 above)
Comment (antonhughes) [reply goodstype/services.py]: done

File: goodstype/urls.py (hunk 7 above)
Comment (antonhughes) [reply goodstype/urls.py]: done

File: goodstype/urls.py (hunk 8 above)
Comment (antonhughes) [reply goodstype/urls.py]: done

File: goodstype/views.py (hunk 10 above)
Comment (antonhughes) [reply goodstype/views.py]: done

File: templates/apply_for_a_licence/goods/index.html (hunk 11 above)
Comment (antonhughes) [reply templates/apply_for_a_licence/goods/index.html]: done

File: templates/apply_for_a_licence/overview.html (hunk 12 above)
Comment (antonhughes) [reply templates/apply_for_a_licence/overview.html]: Done

File: goodstype/views.py (hunk 10 above)
Comment (antonhughes) [reply goodstype/views.py]: This breaks the code, happy to look back on this and see whats going on. maybe it should be 201 for creation?

File: goods/views.py [hunk 13]
```diff
@@ -33,6 +33,20 @@ def post(self, request, **kwargs):
...
+    def get(self, request, **kwargs):
+        return form_page(request, forms.form)
+
+    def post(self, request, **kwargs):
+        data, status_code = post_goods(request, request.POST)
+
+        if status_code == 400:
+            return form_page(request, forms.form, request.POST, errors=data['errors'])
+
+        print(kwargs)
```
Comment (jfaracik) [inline goods/views.py]: print statement

File: goodstype/urls.py [hunk 14]
```diff
@@ -0,0 +1,13 @@
+from django.urls import path
+
+from . import views
+
+app_name = 'goods_type'
+urlpatterns = [
+    # ex: /goodstype/
+    path('<uuid:pk>', views.GoodsType.as_view(), name='goods_type'),
+    # ex: /goodstype/43a88949-5db9-4334-b0cc-044e91827451
+    # path('<uuid:pk>', views.case, name='goodstype'),
```
Comment (jfaracik) [inline goodstype/urls.py]: commented out?

File: goodstype/views.py [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+        return redirect(reverse_lazy('goods:goods'))
+
+
+class DraftAddGoodsType(TemplateView):
+    def get(self, request, **kwargs):
+        return form_page(request, forms.form)
+
+    def post(self, request, **kwargs):
+        copied_post = request.POST.copy()
+        copied_post['content_type'] = DRAFT_CONTENT_TYPE_ID
```
Comment (jfaracik) [inline goodstype/views.py]: This could be finicky when adding tables, should find a better solution

File: goods/views.py (hunk 13 above)
Comment (antonhughes) [reply goods/views.py]: done

File: goodstype/urls.py (hunk 14 above)
Comment (antonhughes) [reply goodstype/urls.py]: done

File: goodstype/views.py (hunk 15 above)
Comment (antonhughes) [reply goodstype/views.py]: Completely agree, we have been working on this but no luck yet, will need to knock our heads together to try and work this out.

Comment (antonhughes) [reply goodstype/views.py]: fixed

File: goodstype/views.py (hunk 10 above)
Comment (antonhughes) [reply goodstype/views.py]: no response to this, closing it down

**Pull request merged**